src/components/deploymentConfig/DeploymentTemplateView.tsx

```diff
@@ -0,0 +1,831 @@
+    isUnSet,
+    disableVersionSelect,
+    charts,
+    selectedChart,
+    selectChart,
+    selectedChartRefId,
+}: {
+    isUnSet: boolean
+    disableVersionSelect?: boolean
+    charts: DeploymentChartVersionType[]
+    selectedChart: DeploymentChartVersionType
+    selectChart: (
+        selectedChart: DeploymentChartVersionType,
+    ) => void | React.Dispatch<React.SetStateAction<DeploymentChartVersionType>>
+    selectedChartRefId: number
+}) => {
+    const uniqueChartsByDevtron = new Map<string, boolean>(),
+        uniqueCustomCharts = new Map<string, boolean>()
+    let devtronCharts = [],
+        customCharts = []
+
+    for (let chart of charts) {
+        const chartName = chart.name
+        if (chart['userUploaded']) {
+            if (!uniqueCustomCharts.get(chartName)) {
+                uniqueCustomCharts.set(chartName, true)
+                customCharts.push(chart)
+            }
+        } else if (!uniqueChartsByDevtron.get(chartName)) {
+            uniqueChartsByDevtron.set(chartName, true)
+            devtronCharts.push(chart)
+        }
+    }
+    devtronCharts = sortObjectArrayAlphabetically(devtronCharts, 'name')
+    customCharts = sortObjectArrayAlphabetically(customCharts, 'name')
+
+    const groupedChartOptions = [
+        {
+            label: 'Charts by Devtron',
+            options: devtronCharts,
+        },
+        {
+            label: 'Custom charts',
+            options: customCharts.length === 0 ? [{ name: 'No options' }] : customCharts,
+        },
+    ]
+    const filteredCharts = selectedChart
+        ? charts.filter((cv) => cv.name == selectedChart.name).sort((a, b) => b.id - a.id)
+        : []
+
+    return (
+        <div
+            style={{
+                display: 'grid',
+                gridTemplateColumns: 'auto auto',
+                columnGap: '16px',
+            }}
+        >
+            <div
+                style={{
+                    display: 'grid',
+                    gridTemplateColumns: '75px 1fr',
+                    alignItems: 'center',
+                }}
+            >
+                <span className="fs-13 fw-4 cn-9">Chart type:</span>
+                {isUnSet ? (
+                    <ReactSelect
+                        options={groupedChartOptions}
+                        isMulti={false}
+                        getOptionLabel={(option) => `${option.name}`}
+                        getOptionValue={(option) => `${option.name}`}
+                        value={selectedChart}
+                        classNamePrefix="chart_select"
+                        isOptionDisabled={(option) => !option.id}
+                        isSearchable={false}
+                        components={{
+                            IndicatorSeparator: null,
+                            DropdownIndicator,
+                            Option,
+                            MenuList: ChartMenuList,
+                        }}
+                        styles={{
```

Comment:
can we please extract out the style to some common place?

Reply:
Done